### ISSUE TITLE
Reopening a file (notebook) should open it as a notebook instead of as a text file

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#180984]
I opened a notebook on vscode.dev, saw a notification to install the appropriate extension, installed the extension, closed and reopened the file and it did not render as a notebook. 

@lramos15 has explained this is because there's metadata stored to indicate how the file was last opened, but I worry beginners would not understand that and would think the extension and/or VS Code are broken

https://user-images.githubusercontent.com/29464607/234714181-27ea620c-57f6-4d35-9585-0b4aa8f860ee.mov

**Comment by bpasero** [comment]
I think the metadata makes sense to be picked when the set of editors that are capable of opening a file have not changed but should probably be dropped when a new editor comes in that was installed.

**Comment by lramos15** [comment]
> I think the metadata makes sense to be picked when the set of editors that are capable of opening a file have not changed but should probably be dropped when a new editor comes in that was installed.

Is this something the resolver controls or something that the history service controls?  I assume you're passing back an explicit editor id here.

**Comment by bpasero** [comment]
Oh, I was not assuming history was involved, but that depends on how this was done:

> closed and reopened the file

If it is literally "Re-open closed editor", then there is a chance that this would not use the new editor type.

**Comment by bpasero** [comment]
Just checked, even though we store the `editorId` for the "Re-open editor" history, we do not apply it when opening, so I do not think there is an issue in that place:

https://github.com/microsoft/vscode/blob/726b7f51e0063b065436be42aaee4853447143d8/src/vs/workbench/services/history/browser/historyService.ts#L632

**Comment by lramos15** [comment]
> Just checked, even though we store the `editorId` for the "Re-open editor" history, we do not apply it when opening, so I do not think there is an issue in that place:

Yes we do, we specifically made this change with https://github.com/microsoft/vscode/issues/117579

**Comment by bpasero** [comment]
Oh I see, that is the other history ;)

**Comment by bpasero** [comment]
I wonder if this should be relaxed a bit, in case a more relevant editor for a file is present, use that if it was not present before? I wonder though how to record that...

**Comment by lramos15** [comment]
I'm not sure how you would be able to tell if a more relevant editor is present, but maybe the resolver could provide some utility that allows you to get the editor id list for a given URI at the time of saving it and you just diff the list. If the list is different something was installed or uninstalled so don't pass the id through, instead let the resolver pick the type.

**Comment by bpasero** [comment]
Here is an idea: if an editor was opened without an explicit editor ID, quick open should actually not store and restore it. I think the original scenario we had in mind was that a user would pick "Re-open editor with..." to pick a specific editor and then we keep that info even in the quick open history.

But if an editor was opened normally, e.g. from explorer without an explicit editor ID, quick open should not persist it.

I am not sure though how to distinguish. Adding Tyler also for quick open.

**Comment by lramos15** [comment]
> I am not sure though how to distinguish.

Yeah this is tough without another property or something passed back because I believe the quick open isn't injected until after the editor is resolved.
